src/org/parosproxy/paros/core/scanner/HostProcess.java

```diff
@@ -614,6 +622,23 @@ public void pluginSkipped(Plugin plugin, String reason) {
      */
     public boolean isSkipped(Plugin plugin) {
         PluginStats pluginStats = mapPluginStats.get(plugin.getId());
+
+        if (this.scannerParam.getMaxRuleDurationInMins() > 0 && plugin.getTimeStarted() != null) {
+        	long endtime = System.currentTimeMillis();
+        	if (plugin.getTimeFinished() != null) {
+        		endtime = plugin.getTimeFinished().getTime();
+        	}
+        	if (endtime - plugin.getTimeStarted().getTime() > 
+        			TimeUnit.MINUTES.toMillis(this.scannerParam.getMaxRuleDurationInMins())) {
+            	return true;

```

Comment:
It could set the reason before returning by calling `pluginSkipped(Plugin, String)`, and the check could be removed from `getSkippedReason(Plugin)`.